### PR TITLE
Add executing transit price computations procedure

### PR DESCRIPTION
**Pull request by vessla**

### CODE REVIEW

File: src/main/java/org/opentripplanner/pricing/transit/ticket/TransitTicket.java [hunk 1]
```diff
@@ -200,7 +200,7 @@ private int getRouteStopConstraintCompliantTime(int ticketShouldBeValidUntil, Li
     private boolean isTicketValid(TransitTripStage tripStage) {
         if (Objects.isNull(tripStage)) return false;
 
-        String agencyId = tripStage.getCurrentRoute().getId().getAgencyId();
```
Comment (vessla) [inline src/main/java/org/opentripplanner/pricing/transit/ticket/TransitTicket.java:203]: Podczas testów na realnym grafie okazało się, że ta wartość nie jest kompatybilna z agency id w GTFS ;)

File: src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java [hunk 2]
```diff
@@ -190,6 +195,17 @@ public static Itinerary generateItinerary(GraphPath path, boolean showIntermedia
         itinerary.timeTraversedInMode = lastState.createTimeTraversedInModeMap();
         itinerary.price = lastState.getTraversalPrice();
 
+        Set<TransitTicket> availableTickets = graph.getAvailableTransitTickets();
+
+        if (Objects.nonNull(availableTickets) && !availableTickets.isEmpty()) {
+            itinerary.getTripStages().addAll(generateTransitTripStages(states));
```
Comment (AdamWiktor) [inline src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java]: `TripStages` could be just a local variable here.

File: src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java [hunk 3]
```diff
@@ -781,6 +797,48 @@ private static void addPlaces(Leg leg, List<State> states, List<Edge> edges, boo
...
+        int currentTripTime = 1;
+        int firstTransitFareStartsAt = -1;
+        Stop currentStop;
+        double distance;
+
+        for (State currentState : states) {
+            Vertex vertex = currentState.getVertex();
+
+            if (vertex instanceof PatternArriveVertex || vertex instanceof PatternDepartVertex) {
+                currentStop = ((TransitVertex) vertex).getStop();
```
Comment (AdamWiktor) [inline src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java]: Please make here new variable `OnboardVertex currentVertex = (OnboardVertex) vertex;` and use it later without casting.

File: src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java [hunk 4]
```diff
@@ -781,6 +797,48 @@ private static void addPlaces(Leg leg, List<State> states, List<Edge> edges, boo
         }
     }
 
+    private static List<TransitTripStage> generateTransitTripStages(List<State> states) {
+        List<TransitTripStage> transitTripStages = new ArrayList<>();
+
+        int currentTripTime = 1;
+        int firstTransitFareStartsAt = -1;
+        Stop currentStop;
```
Comment (AdamWiktor) [inline src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java:804]: Please move declaration of those two variables `currentStop` and `distance` to their definitions (they are used as inner variables inside `if` blocks and not shared)

**Review by AdamWiktor — APPROVED** [review]

**Pull request merged**